src/WebJobs.Script/Description/FunctionInvokerBase.cs

```diff
@@ -23,16 +23,11 @@ public abstract class FunctionInvokerBase : IFunctionInvoker, IDisposable
         private bool _disposed = false;
         private IDisposable _fileChangeSubscription;
 
-        internal FunctionInvokerBase(ScriptHost host, FunctionMetadata functionMetadata)
-            : this(host, functionMetadata, new FunctionLogger(host, functionMetadata.Name))
```

Comment:
Removing the ctor that allows this disposable object to be passed in, so we can assume we control the lifetime and handle disposal